src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1098,6 +1186,146 @@ export default class ComparisonStore {
+                                            acc.push({
+                                                ...group,
+                                                count: filteredSamples.length,
+                                                samples: filteredSamples,
+                                                description: `samples in ${group.name}`,
+                                            });
+                                        }
+                                        return acc;
+                                    },
+                                    []
+                                );
+                            } else {
+                                return [];
+                            }
+                        }
+                    )
+                );
+            },
+        }
+    );
+
+    readonly genericAssayEnrichmentDataRequestGroupsGroupByGenericAssayTypes = remoteData(
+        {
+            await: () => [
+                this
+                    .genericAssayEnrichmentAnalysisGroupsGroupByGenericAssayType,
+                this
+                    .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType,
+            ],
+            invoke: () => {
+                return Promise.resolve(
+                    _.mapValues(
+                        this
+                            .genericAssayEnrichmentAnalysisGroupsGroupByGenericAssayType
+                            .result!,
+                        (
+                            genericAssayEnrichmentAnalysisGroups,
+                            genericAssayType
+                        ) => {
+                            return genericAssayEnrichmentAnalysisGroups.map(
+                                group => {
+                                    const molecularProfileCaseIdentifiers = group.samples.map(
+                                        sample => ({
+                                            caseId: sample.sampleId,
+                                            molecularProfileId: this
+                                                .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType
+                                                .result![genericAssayType][
+                                                sample.studyId
+                                            ].molecularProfileId,
+                                        })
+                                    );
+                                    return {
+                                        name: group.name,
+                                        molecularProfileCaseIdentifiers,
+                                    };
+                                }
+                            );
+                        }
+                    )
+                );
+            },
+        }
+    );
+
+    readonly genericAssayEnrichmentDataGroupByGenericAssayType = remoteData({
```

Comment:
can you please explain what this is?

Reply:
This function get the enrichment data map (keyed by `genericAssayType`). After we have this, we can use `genericAssayType` to get data for each comparison sub-tab.

Reply:
thanks. how about `gaEnrichmentDataByAssayType`?

Reply:
Yes!